src/main/java/uk/gov/companieshouse/officerssearch/subdelta/kafka/ResourceChangedDataDeserialiser.java

```diff
@@ -25,7 +27,7 @@ public ResourceChangedData deserialize(String topic, byte[] data) {
                     ResourceChangedData.class);
             return reader.read(null, decoder);
         } catch (IOException | AvroRuntimeException e) {
-            LOGGER.error("Error deserialising message.", e);
+            LOGGER.error("Error deserialising message.", e, DataMapHolder.getLogMap());
```

Comment:
The deserialiser is called early in the call chain and before the `StructuredLoggingKafkaListenerAspect` so the `DataMapHolder` will not be initialised. This means the `request_id` will still have a value of "uninitialised", which may be confusing.
My solution was not omit the `DataMapHolder.getLogMap()` bit completely

Reply:
Addressed in latest commit